src/main/java/org/qualityannotate/quality/sonarqube/SonarqubeQualityTool.java

```diff
@@ -45,26 +55,41 @@ GlobalMetrics getGlobalMetrics() {
                 String.join(",", config.globalMetricTypes()));
         Map<String, String> metrics = new HashMap<>();
         for (Metric metric : componentMeasures.getMetrics()) {
-            Optional<Measure> measureOpt = componentMeasures.getComponent()
-                                                            .getMeasure(metric.getKey());
+            Optional<Measure> measureOpt = componentMeasures.getComponent().getMeasure(metric.getKey());
             if (measureOpt.isPresent()) {
                 Measure measure = measureOpt.get();
-                String value = measure.getPeriod() == null ? measure.getValue() : measure.getPeriod()
-                                                                                         .value();
-                if (metric.getKey()
-                          .endsWith("coverage")) {
+                String value = measure.getPeriod().map(Measure.Period::value).orElse(measure.getValue());
+                if (value == null) {
+                    value = "";
+                }
+                if (metric.getKey().endsWith("coverage")) {
                     try {
                         value = String.format("%.2f%%", Double.parseDouble(value));
                     } catch (NumberFormatException e) {
                         // ignore
                     }
+                } else {
+                    try {
+                        value = String.format("%.3f", Double.parseDouble(value));
+                        value = value.replaceAll("\\.000$", "");
+                    } catch (NumberFormatException e) {
+                        // ignore
+                    }
                 }
                 metrics.put(metric.getName(), value);
             }
```

Comment:
⚠️ maintainability: This block of commented-out lines of code should be removed. [details](https://sonarcloud.io/organizations/quyt/rules?open=java:S125&rule_key=java:S125)